src/services/permanence-conseillers/permanence-conseillers.class.js

```diff
@@ -0,0 +1,91 @@
+const { Service } = require('feathers-mongodb');
+const { Conflict } = require('@feathersjs/errors');
+const logger = require('../../logger');
+
+const {
+  canActivate,
+  authenticationGuard,
+  authenticationFromRequest,
+  rolesGuard,
+  userIdFromRequestJwt,
+  Role,
+  abort
+} = require('../../common/utils/feathers.utils');
+
+const { userAuthenticationRepository } = require('../../common/repositories/user-authentication.repository');
+const { updatePermanenceToSchema } = require('./permanence/utils/update-permanence.utils');
+const { getPermanenceByConseiller, createPermanence, setPermanence } = require('./permanence/repositories/permanence-conseiller.repository');
+
+exports.PermanenceConseillers = class Sondages extends Service {
+  constructor(options, app) {
+    super(options);
+
+    app.get('mongoClient').then(db => {
+      this.Model = db.collection('permanences');
+    });
+
+    app.get('/permanence-conseillers/conseiller/:id', async (req, res) => {
+
+      const db = await app.get('mongoClient');
+      const user = await userAuthenticationRepository(db)(userIdFromRequestJwt(req));
+      const conseillerId = req.params.id;
+
+      canActivate(
+        authenticationGuard(authenticationFromRequest(req)),
+        rolesGuard(user._id, [Role.Conseiller], () => user)
+      ).then(async () => {
+        await getPermanenceByConseiller(db)(conseillerId).then(permanence => {
+          res.send({ permanence });
+        }).catch(error => {
+          app.get('sentry').captureException(error);
+          logger.error(error);
+          res.status(404).send(new Conflict('La recherche de permanence a échouée, veuillez réessayer.').toJSON());
+        });
+
+      }).catch(routeActivationError => abort(res, routeActivationError));
+
+    });
+    app.post('/permanence-conseillers/create', async (req, res) => {
+      const db = await app.get('mongoClient');
+      const connection = app.get('mongodb');
+      const database = connection.substr(connection.lastIndexOf('/') + 1);
+      const query = updatePermanenceToSchema(req.body.permanence);
+      const user = await userAuthenticationRepository(db)(userIdFromRequestJwt(req));
+
+      canActivate(
+        authenticationGuard(authenticationFromRequest(req)),
+        rolesGuard(user._id, [Role.Conseiller], () => user)
+      ).then(async () => {
+        await createPermanence(db)(query, database).then(() => {
+          res.send({ isCreated: true });
+        }).catch(error => {
+          app.get('sentry').captureException(error);
+          logger.error(error);
+          res.status(409).send(new Conflict('La création de permanence a échoué, veuillez réessayer.').toJSON());
+        });
+      }).catch(routeActivationError => abort(res, routeActivationError));
+
+    });
+
+    app.post('/permanence-conseillers/:id', async (req, res) => {
```

Comment:
S'il s'agit d'une mise à jour totale de la ressource, est-ce que cette syntaxe serait valide ?
```suggestion
    app.put('/permanence-conseillers/:id', async (req, res) => {
```